core/templates/pages/contributor-dashboard-page/modal-templates/certificate-download-modal.component.ts

```diff
@@ -0,0 +1,296 @@
+})
+export class CertificateDownloadModalComponent {
+  @Input() suggestionType!: string;
+  @Input() username!: string;
+  @Input() languageCode!: string | null;
+  fromDate!: string;
+  toDate!: string;
+  errorMessage!: string;
+  errorsFound = false;
+  certificateDownloading = false;
+  datesSelected = false;
+
+  CERTIFICATE_WIDTH: number = 1500;
+  CERTIFICATE_HEIGHT: number = 1300;
+  CERTIFICATE_MID_POINT: number = 750;
+  CERTIFICATE_INNER_BOX_WIDTH: number = 1300;
+  CERTIFICATE_INNER_BOX_HEIGHT: number = 1100;
+  CANVAS_ROOT_COORDINATE: number = 0;
+  INNER_BOX_COORDINATE: number = 100;
+  LOGO_X_COORDINATE: number = 650;
+  LOGO_Y_COORDINATE: number = 150;
+  LOGO_WIDTH: number = 200;
+  LOGO_HEIGHT: number = 80;
+  SIGNATURE_BASE_COORDINATE: number = 400;
+  DATE_BASE_COORDINATE: number = 1100;
+  SIGNATURE_LINE_STARTING_POSITION: number = 300;
+  SIGNATURE_LINE_ENDING_POSITION: number = 500;
+  DATE_LINE_STARTING_POSITION: number = 1000;
+  DATE_LINE_ENDING_POSITION: number = 1200;
+
+  constructor(
+    private readonly activeModal: NgbActiveModal,
+    private contributionAndReviewService: ContributionAndReviewService) {
+  }
+
+  close(): void {
+    this.activeModal.close();
+  }
+
+  downloadCertificate(): void {
+    this.errorsFound = false;
+    this.errorMessage = '';
+    if (
+      !this.fromDate ||
+      !this.toDate ||
+      new Date(this.fromDate) >= new Date(this.toDate)
+    ) {
+      this.errorsFound = true;
+      this.errorMessage = 'Invalid date range.';
+      return;
+    }
+    if (new Date() < new Date(this.toDate)) {
+      this.errorsFound = true;
+      this.errorMessage = 'Please select a \'To\' date that is earlier than ' +
+        'today\'s date';
+      return;
+    }
+    this.certificateDownloading = true;
+    this.contributionAndReviewService.downloadContributorCertificateAsync(
+      this.username,
+      this.suggestionType,
+      this.languageCode,
+      this.fromDate,
+      this.toDate
+    ).then((response: ContributorCertificateResponse) => {
+      this.createCertificate(response);
+      this.certificateDownloading = false;
+    }).catch(() => {
+      this.errorsFound = true;
+      this.certificateDownloading = false;
+      this.errorMessage = (
+        'Not able to download contributor certificate');
+    });
+  }
+
+  disableDownloadButton(): boolean {
+    return this.fromDate === undefined || this.toDate === undefined;
+  }
+
+  createCertificate(response: ContributorCertificateResponse): void {
+    const canvas = document.createElement('canvas');
+    const currentDate = new Date();
+    const dateOptions: Intl.DateTimeFormatOptions = {
+      year: 'numeric',
+      month: 'long',
+      day: 'numeric'
+    };
+    // Textual parts are starting when y coordinate is equals to 350.
+    let linePosition = 350;
+    const image = new Image(this.LOGO_WIDTH, this.LOGO_HEIGHT);
+    image.src = '/assets/images/contributor_dashboard/oppia-logo.jpg';
```

Comment:
Should this path be placed into a constant?

Reply:
Done